homeassistant/components/generic/config_flow.py

```diff
@@ -570,10 +584,16 @@ async def ws_start_preview(
     _LOGGER.debug("Generating websocket handler for generic camera preview")
 
     flow_id = msg["flow_id"]
-    flow = cast(
-        GenericIPCamConfigFlow,
-        hass.config_entries.flow._progress.get(flow_id),  # noqa: SLF001
-    )
+    if msg.get("flow_type", "config_flow") == "config_flow":
+        flow = cast(
+            GenericIPCamConfigFlow,
+            hass.config_entries.flow._progress.get(flow_id),  # noqa: SLF001
+        )
+    else:  # (flow type == "options flow")
+        flow = cast(
+            GenericIPCamConfigFlow,
```

Comment:
Is this meant to be `GenericOptionsFlowHandler`?

Reply:
I think I tried it that way initially but it did not work. 

Will have to retry when back at a proper computer (this evening or tomorrow).

Reply:
I retried and was getting a mypy error, but I've fixed it now in
a8c4278b